bundles/org.eclipse.ui/src/org/eclipse/ui/internal/SearchDecoration.java

```diff
@@ -68,14 +73,39 @@ private static String getValidationError(String regex) {
 			return ""; //$NON-NLS-1$
 		} catch (PatternSyntaxException e) {
 			String message = e.getLocalizedMessage();
+			StringBuilder sBuilder = new StringBuilder();
 
-			// Only preserve the first line of the original error message.
 			int i = 0;
 			while (i < message.length() && "\n\r".indexOf(message.charAt(i)) == -1) { //$NON-NLS-1$
 				i++;
 			}
+			int j = i + 2;
+			while (j < message.length() && "\n\r".indexOf(message.charAt(j)) == -1) { //$NON-NLS-1$
+				j++;
+			}
```

Comment:
It's odd to see \n\r when windows uses \r\n...

Reply:
It works because it matches either `\n` or `\r`, but I agree: it's odd.

Reply:
I don't know how "foobar".indexOf.. can match anything except exactly "foobar".  In any case, whether it works or not  not evident because there is no test that it works on any OS...

Reply:
Perhaps `x.replaceAll("(?s)[\r\n].*", "")` is a more concise way to trim off all but the first line.  One make compile the pattern with Pattern.DOT_ALL to make it more effecient.

Reply:
Fixed that with @Wittmaxi ´s suggestion. That way we don´t even need the while loop!!